Sources/TuistAutomation/Utilities/TargetBuilder.swift

```diff
@@ -69,7 +69,10 @@ public final class TargetBuilder: TargetBuilding {
         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
         xcodeBuildController: XcodeBuildControlling = XcodeBuildController(),
         xcodeProjectBuildDirectoryLocator: XcodeProjectBuildDirectoryLocating = XcodeProjectBuildDirectoryLocator(),
-        simulatorController: SimulatorControlling = SimulatorController()
+        simulatorController: SimulatorControlling = SimulatorController(
+            system: System.shared,
+            devEnvironment: DeveloperEnvironment.shared
```

Comment:
nit: I'd lean to naming the property `developerEnvironment` since the name of the class is not abbreviated, too

Reply:
`DeveloperEnvironment` is very generic to me, why don't we just merge it to `System`?
It currently has 2 properties:
- `var architecture: MacArchitecture { get }` IMO it fits better into System anyway
- `derivedDataDirectory`: more borderline to move it to System, but I wouldn't' keep a separate thing just for this

Reply:
I lean towards fewer of those utilities since it makes things confusing and adds extra overhead to our APIs. I'd perhaps merge `Environment` and `DevEnvironment` into just `Environment` as a read-only interface to get values. And keep the scope of `system` around running system processes.

Reply:
I have a proposal here that could use more input on how to slice these things up: https://github.com/tuist/tuist/discussions/6380.  

`SimulatorController` only cares about the architecture so it should probably just consume an `ArchitactureProviding` protocol which could be fulfilled by whatever implementation we want.  It could be `System` but should *not* be exposed via the current `Systeming` protocol.  Many of these areas have accumulated functionality because it was easier to add than create a new singleton.    Most of our usages are one or two properties from `Environment` so while it's convenient that it's a single singleton folks can reach for, the behavior that's provided by the protocol is pretty muddied.

For the purposes of this PR, I'd like to just focus on the access patterns and `devEnvironment` can just be removed in preference for the approach we land on from that discussion in a future PR.

Reply:
I'm very onboard with this plan 👍🏼